de-lib/src/main/java/org/iplantc/de/apps/client/presenter/submit/SubmitAppForPublicPresenter.java

```diff
@@ -111,14 +139,45 @@ public void go(HasOneWidget container, App selectedApp, AsyncCallback<String> ca
     @Override
     public void onSubmit() {
         if (view.validate()) {
-            publishApp(view.getPublishAppRequest());
+            publishApp(getPublishAppRequest());
         } else {
             AlertMessageBox amb = new AlertMessageBox(appearance.warning(),
                                                       appearance.completeRequiredFieldsError());
             amb.show();
         }
     }
 
+    PublishAppRequest getPublishAppRequest() {
+        PublishAppRequest appRequest = appAutoBeanFactory.publishAppRequest().as();
+        App selectedApp = view.getSelectedApp();
+
+        appRequest.setSystemId(selectedApp.getSystemId());
+        appRequest.setId(selectedApp.getId());
+        appRequest.setName(view.getAppName());
+        appRequest.setDescription(view.getAppDescription());
+        appRequest.setAvus(getAppAvus());
+        appRequest.setReferences(view.getReferenceLinks());
+        appRequest.setDocumentation(view.getMarkDownDocs());
+
+        return appRequest;
+    }
+
+    private List<Avu> getAppAvus() {
+        List<Avu> avus = Lists.newArrayList();
+        for (OntologyHierarchy model : view.getCategoryTree().getCheckedSelection()) {
+            avus.add(ontologyUtil.convertHierarchyToAvu(model));
+        }
+        for (Group community: view.getCommunityTree().getCheckedSelection()) {
+            Avu avu = avuAutoBeanFactory.getAvu().as();
+            avu.setAttribute(Group.COMMUNITY_ATTR);
```

Comment:
This would be better as a config, rather than hard-coded.

Reply:
😖 Oh right, you mentioned that yesterday. I'll update